### ISSUE TITLE
Path menu: add 'Keep original object(s)" option

### ISSUE THREAD

**Issue by rodlie** [friction2d/friction#295]
As discussed in #294 , it would be nice to have an option to delete the original objects when doing a path action (like object to path, difference etc).

The behavior today it to keep everything.

**Comment by Triffen** [comment]
I agree. I can't think of a case where I'd want it to stay, so I'd totally prefer if the object turned into path instead, while keeping its name, position in the tree, keyframes, properties etc

**Comment by rodlie** [comment]
Thanks for reminding me :)

> while keeping its name, position in the tree, keyframes, properties etc

Ok, but the name thing might be a problem(?)

Example:

- RectangleBox 1
- RectangleBox 0
- RectangleBox

Select RectangleBox 0 and click "object to path"

I did the code modifications needed, the new function deletes *(nothing is actually deleted, it's in the undo history)* the old rectangle and renames the new path to "RectangleBox 0" and move it to the correct index:

- RectangleBox 1
- RectangleBox 0 *(this is now a path object)*
- RectangleBox

All fine, but if I undo we end up with:

- RectangleBox 1
- RectangleBox 2 *(now back to a rectangle again)*
- RectangleBox

This is because the (forward) undo history already have a "RectangleBox 0" (the path we created).

This got a bit complicated, I hope you understand what I mean :)

**Comment by Triffen** [comment]
I think it's fine like this, as I don't expect people undoing it often, and in that rare case they can always change the name.

**Comment by rodlie** [comment]
>I think it's fine like this,

Not really, if you undo then the original object change name and you break stuff that depends on that name (expressions etc).

It's better to change the name of the new object to avoid breaking anything (if my rectangle is used in an expression we break that by introducing a new object with the same name that's not a rectangle).

**Comment by rodlie** [comment]
My suggestion is to append "Path" to the new object name(?).

@nuxttux @pgilfernandez @samytichadou

**Comment by Triffen** [comment]
Fine as well. As long as I still know what it is
